src/Distribution/Nixpkgs/Meta.hs

```diff
@@ -21,21 +21,37 @@ import Prelude hiding ( (<>) )
 #endif
 import Control.DeepSeq
 import Control.Lens
+import Data.Either ( isLeft, fromLeft, fromRight )
 import Data.Set ( Set )
 import qualified Data.Set as Set
 import Distribution.Nixpkgs.License
 import Distribution.System
 import GHC.Generics ( Generic )
 import Language.Nix.Identifier
+import Language.Nix.Path
 import Language.Nix.PrettyPrinting
 
+-- | Representation of platform(s) as supported by nixpkgs:
+--
+--     * 'Left' represents the name of a platform list as
+--       found in @lib.platforms@. E. g. at the time of
+--       writing @Left "darwin"@ would represent the
+--       platform tuples @x86_64-darwin@, @aarch64-darwin@,
+--       @i686-darwin@ and @armv7a-darwin@. This is subject
+--       to change as nixpkgs updates @lib.platforms@, of
+--       course.
+--     * 'Right' indicates a single platform tuple
+--       represented using Cabal's 'Platform'.
+type NixpkgsPlatform = Either Identifier Platform
```

Comment:
Could you update this comment to add a quick explanation on why we need to support both these platform/identifier styles?

Reply:
Done in 73104f558c69872ed2dd5ee2a8a80c51077a0ddf.

Reply:
Yeah, that makes sense! Thanks!